src/turnkeyml/build/stage.py

```diff
@@ -50,6 +51,11 @@ def _name_is_file_safe(name: str):
 
 
 class Stage(abc.ABC):
+    class Status(enum.Enum):
+        NOT_STARTED = "not_started"
+        COMPLETED = "completed"
+        INCOMPLETE = "incomplete"
```

Comment:
We now have 3 Status classes that are very similar 😅 

```
class Stage.Status(enum.Enum):
        NOT_STARTED = "not_started"
        COMPLETED = "completed"
        INCOMPLETE = "incomplete"
```
and
```
class build.Status(enum.Enum):
    NOT_STARTED = "not_started"
    PARTIAL_BUILD = "partial_build"
    BUILD_RUNNING = "build_running"
    COMPLETED_BUILD = "completed_build"
    FAILED_BUILD = "failed_build"
```
and
```
class filesystem.FunctionStatus:
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    KILLED = "killed"
```

I'm sorry my original comment was not clear enough. What I meant was to only have a single status class for the whole codebase that was:

```
class build.FunctionStatus
    INCOMPLETE = "incomplete"
    NOT_STARTED = "not_started"
    COMPLETED = "completed"
    FAILED = "failed"
    KILLED = "killed"
```

Let me know if you want me to jump in on this...

Reply:
Implemented!